src/mono/mono.proj

```diff
@@ -477,12 +478,18 @@
       <_MonoRuntimeArtifacts Condition="'$(MonoBundleLLVMOptimizer)' == 'true'" Include="$(MonoLLVMDir)\bin\opt">
         <Destination>$(RuntimeBinDir)\opt</Destination>
       </_MonoRuntimeArtifacts>
-      <_MonoRuntimeArtifacts Condition="'$(MonoAOTBundleLLVMOptimizer)' == 'true'" Include="$(MonoAOTLLVMDir)\bin\llc">
+      <_MonoRuntimeArtifacts Condition="'$(MonoAOTBundleLLVMOptimizer)' == 'true' and '$(OS)' != 'Windows_NT'" Include="$(MonoAOTLLVMDir)\bin\llc">
         <Destination>$(RuntimeBinDir)cross\llc</Destination>
       </_MonoRuntimeArtifacts>
-      <_MonoRuntimeArtifacts Condition="'$(MonoAOTBundleLLVMOptimizer)' == 'true'" Include="$(MonoAOTLLVMDir)\bin\opt">
+      <_MonoRuntimeArtifacts Condition="'$(MonoAOTBundleLLVMOptimizer)' == 'true' and '$(OS)' == 'Windows_NT'" Include="$(MonoAOTLLVMDir)\bin\llc.exe">
+        <Destination>$(RuntimeBinDir)cross\llc.exe</Destination>
```

Comment:
an easier approach I've seen elsewhere is something like this:
```xml
<_MonoRuntimeArtifacts Condition="'$(MonoAOTBundleLLVMOptimizer)' == 'true'" Include="$(MonoAOTLLVMDir)\bin\llc$(ExeExtension)">
  <Destination>$(RuntimeBinDir)cross\llc$(ExeExtension)</Destination>
```

and then setting ExeExtension to .exe only on Windows.